### PR TITLE
feat: add --no-clobber-old-sections switch

### PR DESCRIPTION
**Pull request by K900**
Works around #520, may be useful for other cursed self-modifying things.

### CODE REVIEW

File: src/patchelf.cc [hunk 1]
```diff
@@ -664,14 +665,16 @@ template<ElfFileParams>
...
-       clobbering previously written new section contents. */
-    for (auto & i : replacedSections) {
-        const std::string & sectionName = i.first;
-        const Elf_Shdr & shdr = findSectionHeader(sectionName);
-        if (rdi(shdr.sh_type) != SHT_NOBITS)
-            memset(fileContents->data() + rdi(shdr.sh_offset), 'Z', rdi(shdr.sh_size));
+    if (clobberOldSections) {
+        /* Overwrite the old section contents with 'Z's.  Do this
+        *before* writing the new section contents (below) to prevent
+        clobbering previously written new section contents. */
```
Comment (vcunat) [inline src/patchelf.cc:671]: Nit: didn't indentation of the comment get broken?  (it changed and personally I don't like it much)

Comment (K900) [reply src/patchelf.cc]: Fixed.

File: src/patchelf.cc [hunk 2]
```diff
@@ -2505,6 +2508,7 @@ static void showHelp(const std::string & progName)
   [--clear-execstack]\n\
   [--set-execstack]\n\
   [--rename-dynamic-symbols NAME_MAP_FILE]\tRenames dynamic symbols. The map file should contain two symbols (old_name new_name) per line\n\
+  [--no-clobber-old-sections]\t\tDo not clobber old section values - only use when executable does self-referential tricks.\n\
```
Comment (Mic92) [inline src/patchelf.cc:2511]: What are self-referential tricks? Can you add more details here?

Comment (K900) [reply src/patchelf.cc:2511]: Basically anything that relies on things remaining at the same offset in the binary.

Comment (Mic92) [inline src/patchelf.cc:2511]:
```suggestion
  [--no-clobber-old-sections]\t\tDo not clobber old section values - only use when executables rely on sections remain at the same offset in the binary.\n\
```

**Comment by K900** [comment]
How's this phrasing?

**Comment by Mic92** [comment]
Better. Could you also update https://github.com/NixOS/patchelf/blob/master/patchelf.1 ?

**Comment by K900** [comment]
Done. Tried to elaborate a bit on the behavior in the man page too.

**Pull request merged**